pylint/checkers/unicode.py

```diff
@@ -0,0 +1,713 @@
+        node: nodes.NodeNG,
+        node_type: str,
+        name: str,
+        confidence: interfaces.Confidence,
+    ) -> None:
+        type_label = HUMAN_READABLE_TYPES[node_type]
+        args = (type_label.capitalize(), name)
+
+        self.add_message("non-ascii-name", node=node, args=args, confidence=confidence)
+        self.linter.stats.increase_bad_name(node_type, 1)
+
+    def _check_name(
+        self,
+        node_type: str,
+        names: Union[str, List[str]],
+        node: Union[nodes.NodeNG, _AsciiOnlyCheckedNode],
+        confidence: interfaces.Confidence = interfaces.HIGH,
+    ) -> None:
+        """check for a name if it's using non ASCII and set message if it does
+
+        Also set the dynamic attribute ``_is_ascii_only`` that is used to
+        determine if a node has been already checked, so we don't have to handle
+        too many edge cases.
+        """
+        if not isinstance(names, list):
+            names = [names]
+
+        current_state = getattr(node, "_is_ascii_only", True)
+        for name in names:
+            if name is None:
+                # For some nodes i.e. *kwargs from a dict, the name will be empty
+                continue
+            non_ascii_match = self._non_ascii_rgx_compiled.match(name)
+            if not (Py37Str(name).isascii() and non_ascii_match is None):
+                # Note that we require the `.isascii` method as it is fast and
+                # handles the complexities of unicode, so we can use simple regex.
+                self._raise_name_warning(
+                    node,
+                    node_type,
+                    name,
+                    confidence,
+                )
+                current_state = False
+
+        node._is_ascii_only = current_state  # pylint: disable=protected-access
+
+    def _recursive_check_names(self, args: Iterable[nodes.AssignName]):
+        """check names in a possibly recursive list <arg>"""
+        for arg in args:
+            if isinstance(arg, nodes.AssignName):
+                self._check_name("argument", arg.name, arg)
+            else:
+                self._recursive_check_names(arg.elts)
+
+    @utils.check_messages("non-ascii-name")
+    def visit_module(self, node: nodes.Module) -> None:
+        # pylint: disable-next=fixme
+        # TODO check how the node name is defined when calling the real checker
+        #      in the test using the astroid manager the node.name attribute corresponds
+        #      to the filename -> so we always have ".py" as result of the following
+        #      function.
+        #      It would be nice if we could check for an ASCII filename, if not done
+        #      already by other checkers...
+        self._check_name("module", node.name.split(".")[-1], node)
+
+    @utils.check_messages("non-ascii-name")
+    def visit_functiondef(
+        self, node: Union[nodes.FunctionDef, nodes.AsyncFunctionDef]
+    ) -> None:
+        confidence = interfaces.HIGH
```

Comment:
The confidence is hard coded and the same all the time, instead of passing it to each function we should be hard coding it in ``_raise_name_warning``

Reply:
removed